src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Diagnostics/RazorTranslateDiagnosticsService.cs

```diff
@@ -497,16 +497,14 @@ private bool TryGetOriginalDiagnosticRange(LspDiagnostic diagnostic, RazorCodeDo
             out originalRange))
         {
             // Couldn't remap the range correctly.
-            // If this isn't an `Error` Severity Diagnostic we can discard it.
-            if (diagnostic.Severity != LspDiagnosticSeverity.Error)
+            // If this is error it's worth at least logging so we know if there's an issue
+            // for mapping when a user reports not seeing an error they thought they should
+            if (diagnostic.Severity == LspDiagnosticSeverity.Error)
             {
-                return false;
+                this._logger.LogDebug($"Dropping diagnostic {diagnostic.Code}:{diagnostic.Message} at csharp range {diagnostic.Range}");
```

Comment:
No problem with this as is, but might be worth considering if this should this be an information or perhaps even warning in future?

Reply:
I'll make it a warning. Good call